src/map/skills/battle_skill.cpp

```diff
@@ -0,0 +1,39 @@
+#include "battle_skill.hpp"
+#include <common/cbasetypes.hpp>
+#include "../battle.hpp"
+#include "../status.hpp"
+#include "../pc.hpp"
+#include "../map.hpp"
+#include <bitset>
+
+// Static helper method implementations
+map_session_data* BattleSkill::getPlayerData(const block_list* bl) {
+    if (bl && bl->type == BL_PC) {
+        return (map_session_data*)bl;
+    }
+    return nullptr;
+}
+
+status_data* BattleSkill::getStatusData(const block_list* bl) {
+    if (bl) {
+        return status_get_status_data(const_cast<block_list&>(*bl));
+    }
+    return nullptr;
+}
+
+status_change* BattleSkill::getStatusChange(const block_list* bl) {
+    return status_get_sc(const_cast<block_list*>(bl));
+}
```

Comment:
These helpers are not really needed.
The respective skill can still use the normal macros/functions for these kind of things.